### PR TITLE
Tests for SPCS Azure

### PR DESCRIPTION
**Pull request by alexrenz**
Add a few tests for SPCS Azure, as introduced by [object_store_ffi#28](https://github.com/RelationalAI/object_store_ffi/pull/28)

### CODE REVIEW

**Review by andrebsguedes — COMMENTED** [review]
Looking good, some small comments.

File: src/mock_server.jl [hunk 1]
```diff
@@ -118,6 +120,31 @@ function construct_stage_info(credentials::AWSCredentials, store::AWS.Bucket, pa
     )
 end
 
+function construct_stage_info(credentials::AzureCredentials, store::Azure.Container, path::String, encrypted::Bool)
+    m = match(r"(https?://.*?)/(.*)", store.baseurl)
+    @assert !isnothing(m)
+    test_endpoint = m.captures[1]
+    rest = split(HTTP.unescapeuri(m.captures[2]), "/")
+    account = rest[1]
+    container = rest[2]
```
[the remark this answers is not in the document]
Comment (alexrenz) [reply src/mock_server.jl]: Looks like `parseAzureAccountContainerBlob` wants to see `core.windows.net`: https://github.com/JuliaServices/CloudBase.jl/blob/9738799d6a6d93c042dd06e3a85e599bbebc34b8/src/azure_sas.jl#L140 :/

File: src/mock_server.jl [hunk 2]
```diff
@@ -118,6 +120,31 @@ function construct_stage_info(credentials::AWSCredentials, store::AWS.Bucket, pa
...
+
+    Dict(
+        "locationType" => "AZURE",
+        "location" => container * "/",
+        "path" => path,
```
[the remark this answers is not in the document]
Comment (alexrenz) [reply src/mock_server.jl]: Yes, that makes sense 👍 Done

File: test/snowflake_stage_exception_tests.jl [hunk 3]
```diff
@@ -1,106 +1,154 @@
...
-            global _stale_config = config
+    using RustyObjectStore: RustyObjectStore, ClientOptions, SnowflakeConfig, AbstractConfig
+    using RustyObjectStore: get_object!, put_object, delete_object
 
-            @testset "Insufficient output buffer size" begin
-                input = "1,2,3,4,5,6,7,8,9,1\n" ^ 5
-                buffer = Vector{UInt8}(undef, 10)
-                @assert sizeof(input) == 100
-                @assert sizeof(buffer) < sizeof(input)
+    function run_basic_test_cases(config::AbstractConfig, provider::Symbol)
```
[the remark this answers is not in the document]
Comment (alexrenz) [reply test/snowflake_stage_exception_tests.jl:7]: Sounds good 👍

File: src/mock_server.jl (hunk 1 above)
[the remark this answers is not in the document]
Comment (andrebsguedes) [reply src/mock_server.jl:129]: Did you try with `parseLocal=true`?

**Review by andrebsguedes — APPROVED** [review]
LGTM! Added one more comment on the url parsing thing but you can take it or leave it.

File: src/mock_server.jl (hunk 1 above)
[the remark this answers is not in the document]
Comment (alexrenz) [reply src/mock_server.jl]: I did, but skimmed the function too quickly when it did not work. But I gave it another shot and now figured out what _actually_ is the problem: the function does not like trailing slashes. I fixed that and now it works :)  Thanks for pushing on this!

[the remark this answers is not in the document]
Comment (alexrenz) [reply src/mock_server.jl]: It does also not like http URLs :D

**Pull request merged**